central/cloudsources/datastore/datastore_impl_postgres_test.go

```diff
@@ -0,0 +1,148 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestCloudSourcesDatastorePostgres(t *testing.T) {
+	suite.Run(t, new(datastorePostgresTestSuite))
+}
+
+type datastorePostgresTestSuite struct {
+	suite.Suite
+
+	readCtx      context.Context
+	writeCtx     context.Context
+	postgresTest *pgtest.TestPostgres
+	datastore    DataStore
+}
+
+func (s *datastorePostgresTestSuite) SetupTest() {
+	s.readCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Integration),
+		),
+	)
+	s.writeCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Integration),
+		),
+	)
+
+	s.postgresTest = pgtest.ForT(s.T())
+	s.Require().NotNil(s.postgresTest)
+	s.datastore = getTestPostgresDataStore(s.T(), s.postgresTest.DB)
+}
+
+func (s *datastorePostgresTestSuite) TearDownTest() {
+	s.postgresTest.Teardown(s.T())
+	s.postgresTest.Close()
+}
+
+func (s *datastorePostgresTestSuite) TestCountCloudSources() {
+	count, err := s.datastore.CountCloudSources(s.readCtx, &v1.Query{})
+	s.NoError(err)
+	s.Zero(count)
+
+	s.addCloudSources(100)
+
+	count, err = s.datastore.CountCloudSources(s.readCtx, &v1.Query{})
+	s.NoError(err)
+	s.Equal(100, count)
+}
+
+func (s *datastorePostgresTestSuite) TestGetCloudSource() {
+	nonExistingID := "00000000-0000-0000-0000-000000000000"
+	cloudSource, err := s.datastore.GetCloudSource(s.readCtx, nonExistingID)
+	s.ErrorIs(err, errox.NotFound)
+	s.Empty(cloudSource)
+
+	cloudSource = fixtures.GetStorageCloudSource()
+	err = s.datastore.UpsertCloudSource(s.writeCtx, cloudSource)
+	s.Require().NoError(err)
+
+	roundtripCloudSource, err := s.datastore.GetCloudSource(s.readCtx, cloudSource.GetId())
+	s.NoError(err)
+	s.Assert().Equal(cloudSource, roundtripCloudSource)
```

Comment:
super nit: let's be consistent and either always use `Assert()` or the e.g. `s.Empty`.